### PR TITLE
Mark `num_finite_buckets` required for cloud logging metric bucket_options

### PR DESCRIPTION
**Pull request by pengq-google**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/12964

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:breaking-change
logging: made `growth_factor`, `num_finite_buckets`, and `scale` required for `google_logging_metric`
```

### CODE REVIEW

File: mmv1/products/logging/Metric.yaml [hunk 1]
```diff
@@ -246,6 +247,7 @@ properties:
               - bucket_options.0.exponential_buckets.0.scale
             description: |
               Must be greater than 0.
+            required: true
           - !ruby/object:Api::Type::Double
             name: growthFactor
             at_least_one_of:
```
Comment (c2thorn) [inline mmv1/products/logging/Metric.yaml:253]: We should remove the requirement for `at_least_one_of` for all 3 of the child fields `numFiniteBuckets`, `growthFactor`, and `scale`

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/logging/Metric.yaml (hunk 1 above)
Comment (c2thorn) [reply mmv1/products/logging/Metric.yaml:253]: This is why the build fails, the two restrictions (required, at least one of) are conflicting

**Comment by c2thorn** [comment]
Since this is a feature for the 5.0.0 release, I've updated the base branch to the 5.0.0 branch. There is a second PR required for this change where we update the 5.0.0 upgrade guide for this breaking change on the `main` branch. Instructions are on https://googlecloudplatform.github.io/magic-modules/develop/make-a-breaking-change/#add-upgrade-guide-entries-to-the-main-branch-of-magic-modules

File: mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func ResourceLoggingProjectSink() *schema.Resource {
 		Type:        schema.TypeBool,
 		Optional:    true,
 		Default:     false,
-		ForceNew:    true,
+                ForceNew:    true,
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go:40]:
```suggestion
		ForceNew:    true,
```

File: mmv1/products/logging/Metric.yaml [hunk 3]
```diff
@@ -240,28 +231,19 @@ properties:
...
+            required: true
           - !ruby/object:Api::Type::Double
             name: growthFactor
-            at_least_one_of:
-              - bucket_options.0.exponential_buckets.0.num_finite_buckets
-              - bucket_options.0.exponential_buckets.0.growth_factor
-              - bucket_options.0.exponential_buckets.0.scale
             description: |
               Must be greater than 1.
+            required: true
```
Comment (c2thorn) [inline mmv1/products/logging/Metric.yaml:241]: just making sure, are _all_ of these fields required? Or only `numFiniteBuckets`?

Comment (pengq-google) [reply mmv1/products/logging/Metric.yaml:241]: Yes. Thanks!

File: mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go:2]:
```suggestion
```

**Review by c2thorn — APPROVED** [review]

**Pull request merged**